lib/recog/fingerprint.rb

```diff
@@ -43,7 +47,17 @@ def initialize(xml)
   # @return [Hash,nil] Keys will be host, service, and os attributes
   def match(match_string)
     # match_string.force_encoding('BINARY') if match_string
-    match_data = @regex.match(match_string)
+    begin
+      match_data = @regex.match(match_string)
+    rescue Encoding::CompatibilityError => e
+      # 'ASCII-8BIT' encoded responses will be caught and fixed here.
+      match_data = @regex.match(match_string.force_encoding('UTF-8'))
```

Comment:
FWIW, I'm hitting exceptions here.  I'm not sure we are guaranteed to get perfectly UFT-8-able data, and this will blow up when it doesn't.